### PR TITLE
Removing dependency to git2

### PR DESCRIPTION
**Pull request by vsiles**
This is related to https://github.com/helsing-ai/buffrs/issues/217

git2 is replaced by:
- in buffrs itself, we now call `git status --porcelain` directly, since `gix` is not able to give us information about untracked files.
- in the test-suite, we use `gix` instead of `git2`

The test suite has been updated to deal with some minor behaviour.

I run `time cargo build` before/after to compare since build time was an issue, here are the results on my laptop (M3):

- before:
```
    Finished dev [unoptimized + debuginfo] target(s) in 57.88s

________________________________________________________
Executed in   57.95 secs    fish           external
   usr time  185.65 secs    0.16 millis  185.65 secs
   sys time   55.63 secs    1.15 millis   55.63 secs
```

 - after:
 ```

    Finished dev [unoptimized + debuginfo] target(s) in 15.59s

________________________________________________________
Executed in   15.66 secs    fish           external
   usr time  102.08 secs    0.18 millis  102.08 secs
   sys time   12.89 secs    1.19 millis   12.89 secs
```

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -75,6 +74,8 @@ assert_cmd = "2.0"
 assert_fs = "1.0"
 axum = { version = "0.7.2", default-features = false, features = ["tokio", "http1"] }
 fs_extra = "1.3"
+gix-tempfile = "13.1.0"
```
Comment (vsiles) [inline Cargo.toml:77]: For some reason without this, the version used was `13.0.0` and test failed to build with a missing function definition.

File: src/command.rs [hunk 2]
```diff
@@ -195,6 +197,51 @@ pub async fn package(directory: impl AsRef<Path>, dry_run: bool) -> miette::Resu
         ))
 }
 
+#[cfg(feature = "git")]
```
Comment (vsiles) [inline src/command.rs:200]: In this function, I tried to emulate the old behavior that would not `bail` if the command failed to run altogether.

**Comment by mara-schulke** [comment]
Hi, is it possible to use gix for this? If not we can remove the feature altogether. Parsing cli outputs is super error prone and I would like to avoid this.

**Comment by vsiles** [comment]
As I mention in the issue discussion, `gix` does not yet allow us to do what `git2` was doing: `gix` does not currently have information about untracked files, so we could only complain if some tracked files have changes.
That being said, I took care of calling `git status --porcelain` because it is made exactly for that:

> --porcelain[=<version>]
> Give the output in an easy-to-parse format for scripts. This is similar to the short output, but will remain stable across Git > versions and regardless of user configuration. See below for details.

That leaves as multiple options:

1.  we remove this check altogether
2. we use `gix` and accept to ignore untracked file (for now at least)
3. we use `git` but don't parse the output. If the output is empty, the repo is clean, otherwise it is not and we print the output as is
4. we trust the `--porcelain` option and parse the output

I don't have a strong opinion here, so I'll implement what you think is best

File: src/command.rs [hunk 3]
```diff
@@ -30,6 +30,8 @@ use async_recursion::async_recursion;
 use miette::{bail, ensure, miette, Context as _, IntoDiagnostic};
 use semver::{Version, VersionReq};
 use std::{env, path::Path, str::FromStr};
+#[cfg(feature = "git")]
+use tokio::process;
```
Comment (tomkarw) [inline src/command.rs:34]: I'd just qualify the import in line 202 and limit the amount of `#[cfg(feature = "git")]` spread across the codebase.

**Comment by tomkarw** [comment]
I agree that using `git status --porcelain` will be robust enough, so I vote for option 4.

That said, I'm fuzzy on why we care about uncommitted changes in the first place. Maybe it's a chance to skim some overreaching features fat.

File: Cargo.toml (hunk 1 above)
Comment (tomkarw) [reply Cargo.toml:77]: What are the steps to reproduce this issue? I commented this line out, run `cargo test`/`cargo test --all-features` and it seems to work.

**Comment by vsiles** [comment]
Yes, I think I'm too new to have enough context here ;) I'm leaving that decision to @mara-schulke and you

**Comment by mara-schulke** [comment]
Okay, so im okay with option 4 *but*: I would not error when the parsing fails. So I would at most emit a warning here that git could not be found / the output is malformed. Currently your implementation raises errors which would potentially abort the call.

If you make it best-effort (say: skip this and emit a warning line to the user) im okay with using `--porcelain`! 😊

**Comment by vsiles** [comment]
Should be good for review now. Thanks for all the input.

**Review by tomkarw — APPROVED** [review]
LGTM, let's merge

**Pull request merged**